pkg/pusher/pusher.go

```diff
@@ -123,6 +134,11 @@ func (s *Service) setChunkAsSynced(ctx context.Context, addr swarm.Address) {
 		s.metrics.ErrorSettingChunkToSynced.Inc()
 	} else {
 		s.metrics.TotalChunksSynced.Inc()
+		ta, err := s.tag.GetByAddress(addr)
+		if err != nil {
+			s.logger.Debugf("pusher: get tag by address %s: %v", addr, err)
+		}
+		ta.Inc(tags.StateSynced)
```

Comment:
If the error is not nil, `ta` is probably nil or not usable. This code may panic or set invalid state if we do not return in case of `err != nil`.